Source/options.cpp

```diff
@@ -792,14 +784,14 @@ GraphicsOptions::GraphicsOptions()
     , showFPS("Show FPS", OptionEntryFlags::None, N_("Show FPS"), N_("Displays the FPS in the upper left corner of the screen."), true)
 {
 	resolution.SetValueChangedCallback(ResizeWindow);
-	fullscreen.SetValueChangedCallback(ResizeWindow);
+	fullscreen.SetValueChangedCallback(SetFullscreenMode);
 #if !defined(USE_SDL1) || defined(__3DS__)
 	fitToScreen.SetValueChangedCallback(ResizeWindow);
 #endif
 #ifndef USE_SDL1
 	upscale.SetValueChangedCallback(ResizeWindow);
-	scaleQuality.SetValueChangedCallback(ReinitializeRenderer);
-	integerScaling.SetValueChangedCallback(ResizeWindow);
+	scaleQuality.SetValueChangedCallback(ReinitializeTexture);
+	integerScaling.SetValueChangedCallback(ReinitializeIntegerScale);

```

Comment:
Doesn't integer scaling also affect the potential window size?
(see `CalculatePreferdWindowSize`/ `ResizeWindow` / `SetFullscreenMode`)

Reply:
You hit the nail on the head. The changes to window size are actually the cause of the problem here. Just take them out and integer scaling works entirely as expected for me. Also, `dx_resize()` is no longer necessary.

Reply:
@AJenbo FYI, it looks to me like `CalculatePreferdWindowSize()` was trying to adjust the resolution to the largest integer multiple of the requested resolution. However, that changes the game's resolution by updating `gnScreenWidth` and `gnScreenHeight` to that integer multiple. I think instead the desired effect is to set the logical size of the renderer to the requested resolution and the window size to the integer multiple in order to get the maximum scale possible for that display mode.

If you like, I can work on a commit for this PR to fix everything that came up based on my testing, and then you can rebase however you like.

Reply:
Thanks that would be appreciated, haven't been able to do much at all today.

Reply:
Alright, I fixed all the issues I could find. I'm submitting my approval since AJenbo can't do it himself, but it would be good if someone else reviews my changes before merging.